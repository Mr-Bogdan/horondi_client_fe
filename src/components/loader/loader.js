import React from 'react';
import { useStyles } from './loader.styles';

export const Loader = (props) => {
  const styles = useStyles(props);

  return (
    <div className={styles.wrapper}>
      <div className={styles.ldsDualRing} />
    </div>
  );
};
