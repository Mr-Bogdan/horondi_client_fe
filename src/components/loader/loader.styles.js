import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
  wrapper: {
    width: '100vw',
    height: '100vh',
    overflow: 'hidden',
    display: 'flex',
    animationName: '',
    animationDuration: '',
    gridColumn: (props) => props.gridColumn || 'none'
  },
  ldsDualRing: {
    display: 'inline-block',
    margin: '0 auto',
    alignItems: 'center',
    alignSelf: 'center',
    '&::after': {
      content: '" "',
      display: 'block',
      borderWidth: (props) => (props.width ? parseInt(props.width) / 10 : 6),
      width: (props) => (props.width ? props.width : '64px'),
      height: (props) => (props.height ? props.height : '64px'),
      borderRadius: '50%',
      borderStyle: 'solid',
      borderColor: `${theme.palette.textColor || 'black'} transparent ${
        theme.palette.textColor || 'black'
      } transparent`,
      animation: '$ldsDualRing 1.2s linear infinite'
    }
  },
  '@keyframes ldsDualRing': {
    '0%': {
      transform: 'rotate(0deg)'
    },
    '100%': {
      transform: 'rotate(360deg)'
    }
  }
}));
