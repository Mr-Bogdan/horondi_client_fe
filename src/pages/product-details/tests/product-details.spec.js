import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import ProductDetails from '../product-details';
import { Loader } from '../../../components/loader/loader';

jest.mock('connected-react-router', () => {
  jest.fn();
});
jest.mock('../product-details.styles', () => ({ useStyles: () => ({}) }));
jest.mock('react-redux');
const mockUseContext = jest.fn().mockImplementation(() => ({
  isLight: true
}));
React.useContext = mockUseContext;

const dispatch = jest.fn();
let storage = {
  currency: 0,
  categoryFilter: [],
  isLoading: false,
  product: null,
  productToSend: {}
};

const match = {
  params: {
    id: 0
  }
};

useDispatch.mockImplementation(() => dispatch);
useSelector.mockImplementation(() => storage);

describe('ProductDetails page test', () => {
  it('Should render ProductDetails', () => {
    const component = shallow(<ProductDetails match={match} />);
    expect(component).toBeDefined();
  });

  it('Should render Loader', () => {
    storage = {
      currency: 0,
      categoryFilter: [],
      isLoading: true,
      product: null,
      productToSend: {}
    };
    const component = shallow(<ProductDetails match={match} />);
    expect(component.exists(Loader)).toBe(true);
  });
});
