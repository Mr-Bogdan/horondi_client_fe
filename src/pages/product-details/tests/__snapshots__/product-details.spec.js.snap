// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product details test Should render product details component 1`] = `
<WithStyles(ForwardRef(Card))
  className=""
>
  <Link
    to="/catalog/:category"
  >
    <Memo(ForwardRef(KeyboardBackspaceIcon))
      color="primary"
    />
  </Link>
  <div>
    <div>
      <ProductSizes
        handleSizeChange={[Function]}
        sizeIsNotSelectedError={false}
      />
    </div>
  </div>
  <Toast />
</WithStyles(ForwardRef(Card))>
`;
